rust/lance-core/src/io/writer.rs

```diff
@@ -123,21 +117,10 @@ impl FileWriter {
     /// Returns [Err] if the schema does not match with the batch.
     pub async fn write(&mut self, batches: &[RecordBatch]) -> Result<()> {
         for batch in batches {
-            // Compare with metadata reset
-            let schema = batch
-                .schema()
-                .as_ref()
-                .clone()
-                .with_metadata(HashMap::new());
-            if self.arrow_schema != schema {
-                return Err(Error::Schema {
-                    message: format!(
-                        "FileWriter::write: schema mismatch: expected: {:?}, actual: {:?}",
-                        self.arrow_schema, schema
-                    ),
-                    location: location!(),
-                });
-            }
+            // Compare, ignore metadata and dictionary
+            //   dictionary should have been checked earlier and could be an expensive check
+            let schema = Schema::try_from(batch.schema().as_ref())?;
+            schema.check_compatible(&self.schema, &SchemaCompareOptions::default())?;
```

Comment:
Anywhere like here where we create a Lance schema from an Arrow schema, we should not check field ids. (This code does that as is, but wanted to note that is different than existing behavior.)